pkg/logs/auditor/auditor.go

```diff
@@ -68,15 +68,18 @@ func New(runPath string, health *health.Handle) *Auditor {
 
 // Start starts the Auditor
 func (a *Auditor) Start() {
+	a.mu.Lock()
 	a.inputChan = make(chan *message.Message, config.ChanSize)
 	a.done = make(chan struct{})
+	a.mu.Unlock()
 	a.registry = a.recoverRegistry()
 	a.cleanupRegistry()
 	go a.run()
 }
 
 // Stop stops the Auditor
 func (a *Auditor) Stop() {
+	a.mu.Lock()
```

Comment:
https://github.com/DataDog/datadog-agent/pull/4860/files#diff-5142632c5dd9990aa6feb0918f9334f7L85 is in the critical path but I think it should not because any method for the run method acquiring the mutex can be called concurrently

Reply:
You could probably create a setup and reset methods which are protected by the mutex

Reply:
I was also thinking to move it in the New() function.

Reply:
make sense. I added the new mutex.